src/modules/ekf2/EKF/ekf.h

```diff
@@ -944,13 +933,12 @@ class Ekf final : public EstimatorInterface
 	bool checkAndFixCovarianceUpdate(const SquareMatrixState &KHP);
 
 	// limit the diagonal of the covariance matrix
-	// force symmetry when the argument is true
-	void fixCovarianceErrors(bool force_symmetry);
+	void constrainStateVariances();
 
-	void constrainStateVar(const IdxDof &state, float min, float max);
+	void forceCovarianceSymmetry();
 
-	// constrain the ekf states
-	void constrainStates();
+	void constrainStateVar(const IdxDof &state, float min, float max);
+	void constrainStateVarLimitRatio(const IdxDof &state, float min, float max, float max_ratio = 100.f);
```

Comment:
100 magic number?

```suggestion
	void constrainStateVarLimitRatio(const IdxDof &state, float min, float max, float max_ratio = 100.f);
```

Reply:
Yes, might be too conservative, but it's hard to tell what's the highest safe value.